crates/bevy_ecs/src/system/commands/apply_commands.rs

```diff
@@ -0,0 +1,38 @@
+use crate::system::{CommandQueue, Commands};
+use crate::world::World;
+
+/// Extension trait used to apply [`Commands`] immediately on a [`World`].
+pub trait ApplyCommands {
+    fn apply_commands<R, F: FnOnce(&World, Commands) -> R>(self, f: F) -> R;
+}
+
+impl ApplyCommands for &mut World {
+    /// Applies some [`Commands`] on this [`World`] immediately.
+    ///
+    /// This function is not an efficient method of applying commands because it requires the creation of a
+    /// dedicated [`CommandQueue`] per call.
+    /// However, this function provides a convenient tool for diagnostics and testing because it allows you to
+    /// invoke commands on a world immediately, without the need for a system.
+    /// Therefore, its use should be reserved for special cases where performance is not a concern.
```

Comment:
should add something here to say what you should do for normal situations. i.e. call methods that act directly on `World`

Reply:
I feel like that's a bit overly verbose. It should be axiomatically known that if you have a `&mut World`, you can modify that world without commands. I'd also argue that methods which act directly on `World` aren't "normal situations". In normal situations `Commands` and systems are used to modify the world, which is why we added the link to `Commands`.